content/Refactoring - Sebastian Herrera.md

```diff
@@ -0,0 +1,144 @@
+# Refactoring
+If you are reading this article I can assume that you are software developer, you have probably both heard the term “refactoring”, and applied it yourself (conscious or not) several times in the code you have written. To begin, let's see the definition:
+
+**Refactoring (noun):
+A change made to the internal structure of software to make it easier to understand and cheaper to modify without changing its observable behavior.**
+
+I became interested in the topic once in my early years as a developer, when I was asked to refactor a piece of code that had plenty of opportunities for improvement. It was a very long method with several steps that performed different things, even with multiple different responsibilities. My first approach to that refactoring was to apply lots of “Extract Method” to that long method, giving those extracted the most representative names that I could find, for them and for the parameters. When I finished doing that, I felt that the code was much better and that I had accomplished the task I was assigned.
+
+Then I presented it to the architect of the project, and he instantly noticed that my refactoring skills were similar to my guitar-playing skills (FYI: I don't play guitar, although I do play air guitar). As any good mentor should do, he saw an opportunity for me to improve, so we paired up, and had about 3 different 1-2 hours long working sessions, designing a new solution together, applying some design patterns here and there.
+
+After reaching to a really much better solution than the original, I really liked what we had done, and I saw a lot of potential in this technique. A code that already worked, now worked the same, but it was much easier to understand and modify whenever it was needed.
+
+Last year, I felt that I definitely could keep improving my skills, so after searching for books recommendations, I reached one very well known: "Refactoring" by Martin Fowler. Now I can say that I understand why it is so recommended. This topic I became interested long ago, now made much more sense to me.
```

Comment:
> This topic I became interested long ago, now made much more sense to me.

This sentence feels a little weird, at least for me. What do you think?

```suggestion
Last year, I felt that I definitely could keep improving my skills, so after searching for book recommendations, I reached one very well known: "Refactoring" by Martin Fowler. Now I can say that I understand why it is so recommended. This topic that interested me a long time ago now makes much more sense to me.
```

Reply:
Not sure on the `that interested me` suggestion. The rest looks good. Thanks.